state/upgrades.go

```diff
@@ -65,6 +65,40 @@ func replaceBsonDField(d bson.D, name string, value interface{}) error {
 	return errors.NotFoundf("field %q", name)
 }
 
+// RenameAddModelPermission renames any permissions called addmodel to add-model.
+func RenameAddModelPermission(st *State) error {
+	coll, closer := st.getRawCollection(permissionsC)
+	defer closer()
+	upgradesLogger.Infof("migrating addmodel permission")
+
+	iter := coll.Find(nil).Iter()
+	defer iter.Close()
+	var ops []txn.Op
+	var doc bson.M
+	for iter.Next(&doc) {
+		fieldVal, ok := doc["access"]
+		if !ok || fieldVal != "addmodel" {
+			continue
+		}
+		id, ok := doc["_id"]
+		if !ok {
+			return errors.New("no id found in permission doc")
+		}
+
+		update := bson.DocElem{"$set", bson.D{{"access", "add-model"}}}
```

Comment:
Why not `bson.D{{""$set", bson.D{{"access", "add-model"}}}}`? It's shorter and saves having to wrap the bson.D later.

Also, there's no need to define this in the loop. Given the update is always the same, just define it above.